zc.recipe.egg_/src/zc/recipe/egg/egg.py

```diff
@@ -166,6 +159,60 @@ def install(self):
 
     update = install
 
+
+_working_set_cache = {}
+
+
+def working_set(
+    distributions,
+    eggs_dir,
+    develop_eggs_dir,
+    offline,
+    newest,
+    links,
+    index,
+    allow_hosts,
+    extra=(),
+):
+    """Helper function to build a working set.
+
+    Results are cached in the `_working_set_cache` dict at module level.
+    The cache key is composed by all the arguments passed to the function.
+    """
+    orig_distributions = list(distributions)
+    cache_key = (
+        tuple(distributions),
```

Comment:
> Can the order of distributions affect the generated working set?

That is a good question. In theory, there should be no difference as long as packages don't try to place modules in the same namespace as other packages and a smart-enough dependency resolver should try to find a set of dependencies for the requested packages no matter the order the packages are requested.

In practice I think the order matters because the resolver in `setuptools` is **not** smart enough and the specific ordering can mean the difference between a dependency resolution that works and one that doesn't, so leave the `distributions` unsorted for now.